lib/spack/spack/binary_distribution.py

```diff
@@ -1330,9 +1349,9 @@ def _build_tarball_in_stage_dir(spec: Spec, out_url: str, stage_dir: str, option
     # create an index.json for the build_cache directory so specs can be
     # found
     if options.regenerate_index:
-        generate_package_index(url_util.join(out_url, os.path.relpath(cache_prefix, stage_dir)))
-
-    return None
+        generate_package_index(
+            url_util.join(out_url, os.path.relpath(cache_prefix, stage_dir)), required=True
```

Comment:
Why `required=True` here, and not on the other call site?

Reply:
The old behavior was to just be totally fine when this failed. I didn't want to break some other part of spack on accident that may be relying on that behavior so I added the flag and only enabled failing in the two places that I was concerned about in this PR.